include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -336,6 +335,8 @@ struct OrtApi {
 
   // Sets the number of threads used to parallelize the execution within nodes
   // A value of 0 means ORT will pick a default
+  // Note: If you've built ORT with OpenMP, this API has no effect on the number of threads used. In this case
```

Comment:
Should this return an error if built with OpenMP to provide a stronger signal?

Or alternatively should the implementation call omp_set_num_threads so things are more consistent?

Reply:
I didn't return an error intentionally thinking that this is changing the behavior of the API. Also thought about using omp_set_num_threads, but it is confusing because not all openmp settings can be controlled by the openmp APIs for e.g. the active/passive wait policy. To avoid confusion, I think it's better to say that if you've built with openmp, use only the openmp env variables to control the threads. Parallel executor is a different story since we use a different threadpool for it anyway.